vmdb/app/models/miq_database.rb

```diff
@@ -100,4 +101,11 @@ def verify_credentials(auth_type=nil, options={})
 
     MiqTask.wait_for_taskid(RegistrationSystem.verify_credentials_queue).task_results if auth_type == :registration
   end
+
+  private
+
+  def registration_server_url_is_valid
+    return if registration_type != "rhn_satellite"  # Only validating Satellite 5 URL
+    errors.add(:registration_server, "expecting http://server.example.com/XMLRPC") unless /^https?:\/\/.+\/XMLRPC$/i === registration_server
```

Comment:
Also, prefer \A and \Z over ^ and $: https://github.com/presidentbeef/brakeman/blob/master/lib/brakeman/checks/check_validation_regex.rb#L3-L9